src/Symfony/Component/Scheduler/Bridge/Google/Worker.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler\Bridge\Google;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Scheduler\Bridge\Google\Task\State;
+use Symfony\Component\Scheduler\Events\TaskExecutedEvent;
+use Symfony\Component\Scheduler\Events\TaskToExecuteEvent;
+use Symfony\Component\Scheduler\Events\WorkerStartedEvent;
+use Symfony\Component\Scheduler\Events\WorkerStoppedEvent;
+use Symfony\Component\Scheduler\Task\TaskInterface;
+use Symfony\Component\Scheduler\WorkerInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * @author Guillaume Loulier <contact@guillaumeloulier.fr>
+ */
+final class Worker implements WorkerInterface
+{
+    /**
+     * @var Client
+     */
+    private $client;
+
+    /**
+     * @var EventDispatcherInterface|null
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var LoggerInterface|null
+     */
+    private $logger;
+
+    public function __construct(Client $client, ?EventDispatcherInterface $eventDispatcher, ?LoggerInterface $logger)
+    {
+        $this->client = $client;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function execute(TaskInterface $task): void
+    {
+        $this->dispatchEvent(new WorkerStartedEvent($this));
+        $this->checkTaskState($task);
+        $this->dispatchEvent(new TaskToExecuteEvent($task));
+
+        $this->client->run($task->get('configuration'));
+
+        $this->dispatchEvent(new TaskExecutedEvent($task, null));
+        $this->dispatchEvent(new WorkerStoppedEvent($this));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function checkTaskState(TaskInterface $task): void
+    {
+        if (State::STATE_UNSPECIFIED === $task->get('state')) {
+            throw new \LogicException('The task state must be defined in order to be executed!');
+        }
+
+        if (\in_array($task->get('state'), [State::PAUSED, State::DISABLED])) {
+            $this->log(sprintf('The following task "%s" is paused|disabled, consider enable it if it should be executed!', $task->getName()));
```

Comment:
i would log the task name in the context of the log, and keep the log message identical

Reply:
Same